### PR TITLE
[playground] Add sessionStorage and some examples

### PR DESCRIPTION
**Pull request by wkazmierczak**
image
image
image
image

### CODE REVIEW

**Review by wkozyra95 — COMMENTED** [review]
save to both session storage and local storage and if you open the card should read from session, but if there is nothing there try local storage, so if you close the only card it still preserve the content

File: docs/src/components/PlaygroundSettingsExamples.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  jsonExample,
+  populateEditorWithExample,
+  closeModal,
+}: ExampleInfoProps) {
+  return (
+    <div className="flex m-4 border-b-solid border-t-solid">
+      <div className="flex-[0_0_20%] min-w-5 text-center font-bold text-xl">{exampleName}</div>
+      <div className="flex-[0_0_65%] px-5 ">{description}</div>
+      <button
+        className="flex-[0_0_15%] rounded-xl font-bold bg-[var(--docsearch-primary-color)] text-lg h-16 text-[var(--ifm-color-emphasis-0)] border-solid border border-[var(--ifm-color-primary)]"
```
[the remark this answers is not in the document]
Comment (wkazmierczak) [reply docs/src/components/PlaygroundSettingsExamples.tsx]: After changes: 
image

File: docs/src/scene/videoCallWithLecturerExample.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Api } from 'live-compositor';
 
-export function defaultJsonScene(): Api.Component {
+export function videoCallWithLecturerExample(): Api.Component {
```
Comment (wkozyra95) [inline docs/src/scene/videoCallWithLecturerExample.ts:3]: leave orginal name, videoCallWithLecturerExample is not a helpful name

File: docs/src/scene/tvPresenterExample.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { Api } from 'live-compositor';
+
+export function tvPresenterExample(): Api.Component {
```
Comment (wkozyra95) [inline docs/src/scene/tvPresenterExample.ts:3]:
```suggestion
export function removeGreenScreenExample(): Api.Component {
```

File: docs/src/components/PlaygroundCodeEditor.tsx [hunk 4]
```diff
@@ -5,10 +5,11 @@ import componentTypesJsonSchema from '../../component_types.schema.json';
 import JSONEditor from '../jsonEditor';
 import { jsonrepair } from 'jsonrepair';
 import Ajv from 'ajv';
+import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
 
 interface PlaygroundCodeEditorProps {
   onChange: (content: object | Error) => void;
-  initialCodeEditorContent: object;
+  codeExample: object;
```
Comment (wkozyra95) [inline docs/src/components/PlaygroundCodeEditor.tsx:12]: why example? Does this value represent example code or content of the editor?

File: docs/src/components/PlaygroundSettingsExamples.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  return (
+    <div>
+      <div className=" flex">
+        <div className="flex-[0_0_20%] flex font-semibold text-center py-3 min-w-5 justify-center items-center">
+          {exampleName}
+        </div>
+        <div className="flex-[0_0_70%] flex text-left py-3 px-5 items-center">{description}</div>
+        <div className="flex-[0_0_10%] flex py-3 justify-center items-center">
+          <button
+            className="rounded-xl font-bold bg-[var(--docsearch-primary-color)] text-lg text-[var(--ifm-color-emphasis-0)] border-solid border border-[var(--ifm-color-primary)]"
```
Comment (wkozyra95) [inline docs/src/components/PlaygroundSettingsExamples.tsx:64]:
```suggestion
            className="rounded-md px-2 font-bold bg-[var(--docsearch-primary-color)] text-lg text-[var(--ifm-color-emphasis-0)] border-solid border border-[var(--ifm-color-primary)]"
```

File: docs/src/components/PlaygroundSettingsExamples.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  closeModal,
+}: ExampleInfoProps) {
+  return (
+    <div>
+      <div className=" flex">
+        <div className="flex-[0_0_20%] flex font-semibold text-center py-3 min-w-5 justify-center items-center">
+          {exampleName}
+        </div>
+        <div className="flex-[0_0_70%] flex text-left py-3 px-5 items-center">{description}</div>
+        <div className="flex-[0_0_10%] flex py-3 justify-center items-center">
```
Comment (wkozyra95) [inline docs/src/components/PlaygroundSettingsExamples.tsx:62]:
```suggestion
        <div className="flex py-3 justify-center items-center">
```
Elements that have constant with should not be "estimated" with the percent of a parent via flex. If you want you can use flex to space that element more on larger screens.

btw I'm not an expert but I think specifying basis as % is a misuse of flex API, checkout what each property is responsible for.

File: docs/src/components/PlaygroundSettingsExamples.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      <ExampleInfo
+        exampleName="Video call"
+        description="Grid layout commonly used to display participants of a video call. One tile has a border rendered with a shader. One tile renders an image instead of a stream."
+        jsonExample={videoCallExample}
+        populateEditorWithExample={populateEditorWithExample}
+        closeModal={closeModal}
+      />
+      <ExampleInfo
+        exampleName="Greenscreen"
+        description="Image of the man (with shader replacing greenscreen) with live-compositor logo in the top-left corner."
```
Comment (wkozyra95) [inline docs/src/components/PlaygroundSettingsExamples.tsx:31]:
```suggestion
        description="Image of a man (with shader replacing greenscreen) with live-compositor logo in the top-left corner."
```

File: docs/src/components/PlaygroundSettingsExamples.tsx (hunk 6 above)
Comment (wkozyra95) [reply docs/src/components/PlaygroundSettingsExamples.tsx:62]: ok other popups have the same problem feel free to ignore this part

File: docs/src/pages/playground.tsx [hunk 8]
```diff
@@ -118,13 +136,22 @@ function Homepage() {
...
+  }, [example]);
+
   return (
     <div className="flex flex-row flex-wrap p-8 lg:h-[calc(100vh-110px)]">
       <div className="flex-1 m-2 border-2 border-gray-400 border-solid rounded-md max-h-full min-w-[300px] min-h-[500px]">
         {showReactEditor ? (
           <PlaygroundReactEditor code={code} onCodeChange={setCode} />
         ) : (
-          <PlaygroundCodeEditor onChange={setScene} initialCodeEditorContent={INITIAL_SCENE} />
+          <PlaygroundCodeEditor onChange={setScene} codeEditorContent={example} />
```
Comment (wkozyra95) [inline docs/src/pages/playground.tsx:154]:
```suggestion
          <PlaygroundCodeEditor onChange={setScene} codeEditorOverrideContent={codeEditorOverrideContent} />
```

File: docs/src/pages/playground.tsx [hunk 9]
```diff
@@ -33,8 +39,18 @@ const INITIAL_REACT_CODE = [
...
+  if (sessionStorageContent) {
+    return sessionStorageContent;
+  } else {
+    return localStorage.getItem('playgroundCodeEditorContent');
+  }
+}
+
 function Homepage() {
   const [scene, setScene] = useState<object | Error>(INITIAL_SCENE);
+  const [example, setExample] = useState<object>(INITIAL_SCENE);
```
Comment (wkozyra95) [inline docs/src/pages/playground.tsx:53]:
```suggestion
  // Is updated when code editor content changes, but changes here are not passed back to the editor.
  const [scene, setScene] = useState<object | Error>(INITIAL_SCENE);
  // Code editor will be updated if this state changes.
  const [codeEditorOverrideContent, setCodeEditorOverrideContent] = useState<object>(INITIAL_SCENE);
```

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**